File: problems/algorithms/date-diff-timestamp/js/response/diff-timestamp.js
function diffMonth(start, end) {
  start = new Date(start);
  end = new Date(end);

  let startYear = start.getFullYear();
  let startMonth = start.getMonth();
  let endYear = end.getFullYear();
  let endMonth = end.getMonth();

  return (endYear - startYear) * 12 + (endMonth - startMonth);
}

export { diffMonth };

File: problems/algorithms/date-diff-timestamp/ruby/response/src/diff-timestamp.rb
require 'date'

class DateDiff
    def DateDiff.compute(start, endDate)
        start = Date.new(start)
        endMonth = Date.new(endDate)

        startYear = Date.strptime(start, '%Y')
        startMonth = Date.strptime(strptime, '%m')
        endYear = Date.strptime(endDate, '%Y')
        endMonth = Date.strptime(endDate, '%m')
        return(endYear - startYear) * 12 + (endMonth - startMonth)
    end
end